### DISCUSSION TITLE
Can we patch our own onSuccess and onError callbacks for a query??

### DISCUSSION THREAD

**Discussion by ashishsurya** [TanStack/query#7483] [Q&A] [answered]
![image](https://github.com/TanStack/query/assets/72222785/af446e9d-8f1a-4116-bf12-106d1939c456)
@TkDodo can we patch our own version of `onSuccess` and `onError` callbacks for a query

**Comment by TkDodo** [comment, marked as answer]
sure, but if you function closes over something from your component / custom hook, like a prop or state, and you use this in multiple components, it's not guaranteed which thing will be "seen". If you have a pure function it will work.

**Reply by ashishsurya** [reply]
Ok so I should not set any state inside the patched callbacks, is it??

**Reply by TkDodo** [reply]
you should not _read_ any props or state inside it. But seeing that this is not clear to you, I would refrain from such an abstraction. There's a reason we removed those ;)

**Reply by ashishsurya** [reply]
Ok, thank you for the feedback

**Reply by ashishsurya** [reply]
![image](https://github.com/TanStack/query/assets/72222785/b2475e7d-1bc3-4370-ba76-cd59476825a2)
Can you confirm me one thing, can I set the state in this way
